components/canvas/canvas_msg.rs

```diff
@@ -0,0 +1,46 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use canvas_paint_task::FillOrStrokeStyle;
+use geom::matrix2d::Matrix2D;
+use geom::point::Point2D;
+use geom::rect::Rect;
+use geom::size::Size2D;
+use std::sync::mpsc::{Sender};
+
+#[derive(Clone)]
+pub enum CanvasMsg {
+    // 2dContext
```

Comment:
Please move this to be a doc comment on `CanvasMsg`:

    /// 2D context.
    #[derive(Clone)]
    pub enum CanvasMsg {
        ...
    }